packaging/create-deb.sh

```diff
@@ -0,0 +1,181 @@
+#!/bin/bash
+#Exit on error
+set -e
+set -x
+
+PPA=release
+CPDIRS=("app" "infrastructure" "test")       
+CPFILES=( ".bowerrc" ".gitattributes" "Gruntfile.js" "karma.conf.js" "LICENSE" "README.md" "bower.json" ".editorconfig"  ".gitignore"      ".jshintrc"     "karma-e2e.conf.js"  "package.json"  ".travis.yml")
+
+#Don't edit below
+
+HOME=`pwd`
+AWK=/usr/bin/awk
+HEAD=/usr/bin/head
+GIT=/usr/bin/git
+SORT=/usr/bin/sort
+DCH=/usr/bin/dch
+PR=/usr/bin/pr 
+SED=/bin/sed
+FMT=/usr/bin/fmt
+PR=/usr/bin/pr
+XARGS=/usr/bin/xargs
+CUT=/usr/bin/cut
+
+
+
+cd $HOME/targets
+TARGETS=(*)
+echo $TARGETS
+cd $HOME
+
+
+
+
+
+RLSVERS=`$GIT tag -l 'v*' | $SED 's/^v//' | $SORT -rV | $HEAD -1`
+PKGVERS=`$GIT tag -l 'ubuntu-*' | $SED 's/^ubuntu-//' | $SORT -rV | $HEAD -1`
+PKGRLS=`$GIT tag -l 'ubuntu-*' | $SED 's/^ubuntu-//' | $SED 's/-.*//' | $SORT -rV | $HEAD -1`
+if [ -n "$PKGVERS"]  || [ $PKGRLS != $RLSVERS ] ; then
+    echo no ubuntu release found
+    VERS="$RLSVERS-1"
+    LASTVERS="$RLSVERS"
+else
+    echo have an existing ubuntu release. bumping it
+    LASTVERS="ubuntu-$PKGVERS"
+    VERS="${RLSVERS}-$((${PKGVERS##*-}+1))"
+fi
+
+
+echo Current ubuntu  verison is $RLSVERS.  Want to create ubuntu release $VERS
+$GIT status
+echo Should we update changelogs, commit everything under 'packaging'  and tag the release as $VERS? [y/n]
+read INCVERS 
+
+echo Should we package only against the formal release v$RLSVERS rather than the latest commit? [y/n]
+read FORMAL
+
+if [[ "$INCVERS" == "y" || "$INCVERS" == "Y" ]];  then
+    TAG="ubuntu-$VERS"
+    COMMITMSG="Release Version $VERS"
+    WIDTH=68
+    URL=$($GIT config --get remote.origin.url | $SED 's/\.git//' | $SED 's/$/\/commmit\//')
+
+
+
+
+    LOGLINES=$($GIT log --oneline $LASTVERS.. | $HEAD -20 | $AWK '{printf " -%s\n --'$URL'%s\n" , $0, $1}')
+
+    FULLCOMMITMSG=$(echo "$COMMITMSG 
+$LOGLINES" |  $XARGS -0 | $AWK '{printf "%-'"$WIDTH.$WIDTH"'s\n" , $0}')
+
+
+    for TARGET in "${TARGETS[@]}"
+    do
+	cd $HOME/targets/$TARGET
+	$DCH -Mv "${VERS}~$TARGET" --distribution "${TARGET}" "${FULLCOMMITMSG}"
+    done
+    cd $HOME
+    
+    $GIT  status
+
+    $GIT add .
+
+    echo "Incrementing version"
+    $GIT commit ./ -m "\"${COMMITMSG}\""
+    $GIT tag $TAG
```

Comment:
We are already tagging releases for our npm releases, could we not make use of those rather?